### PR TITLE
base classes can now have x-ms-discriminator-value extension on them

### PR DESCRIPTION
**Pull request by amarzavery**

### CODE REVIEW

**Comment by amarzavery** [comment]
@lmazuel - Please take a look.

File: src/modeler/AutoRest.Swagger.Tests/Resource/Swagger/swagger-x-ms-discriminator-value.json [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+  },
+  "definitions": {
+    "pet": {
+      "required": [
+        "id",
+        "name",
+        "type"
+      ],
+      "discriminator": "type",
+      "x-ms-discriminator-value": "Microsoft.Models.MSPet",
```
Comment (amarzavery) [inline src/modeler/AutoRest.Swagger.Tests/Resource/Swagger/swagger-x-ms-discriminator-value.json]: @lmazuel - Base model "pet" has the discriminator and the x-ms-discriminator-value extension.

File: src/modeler/AutoRest.Swagger.Tests/SwaggerModelerTests.cs [hunk 2]
```diff
@@ -250,6 +250,32 @@ public void TestcodeModelWithInheritance()
...
+                {
+                    Namespace = "Test",
+                    Input = Path.Combine(Core.Utilities.Extensions.CodeBaseDirectory, "Resource", "Swagger", "swagger-x-ms-discriminator-value.json")
+                };
+                var modeler = new SwaggerModeler();
+                var codeModel = modeler.Build();
+
+                Assert.NotNull(codeModel);
+                Assert.Equal("Pet", codeModel.ModelTypes.First(m => m.Name == "Pet").Name);
+                Assert.Equal("Microsoft.Models.MSPet", codeModel.ModelTypes.First(m => m.Name == "Pet").Extensions["x-ms-discriminator-value"]);
```
Comment (amarzavery) [inline src/modeler/AutoRest.Swagger.Tests/SwaggerModelerTests.cs]: @lamzuel - this test verifies that the base model "pet" has the extension on it. 

I generated the insights client locally by modifying the swagger (Base model RuleDataSource had the x-ms-discriminator-value extension applied to it) and here is the updated discriminator in NodeJS.

```javascript
exports.discriminators = {
  'RuleCondition' : exports.RuleCondition,
  >>>>>>>> 'Microsoft.Azure.Management.Insights.Models.RuleDataSource' : exports.RuleDataSource,
  'RuleDataSource.Microsoft.Azure.Management.Insights.Models.RuleMetricDataSource' : exports.RuleMetricDataSource,
  'RuleDataSource.Microsoft.Azure.Management.Insights.Models.RuleManagementEventDataSource' : exports.RuleManagementEventDataSource,
  'RuleCondition.Microsoft.Azure.Management.Insights.Models.ThresholdRuleCondition' : exports.ThresholdRuleCondition,
  'RuleCondition.Microsoft.Azure.Management.Insights.Models.LocationThresholdRuleCondition' : exports.LocationThresholdRuleCondition,
  'RuleCondition.Microsoft.Azure.Management.Insights.Models.ManagementEventRuleCondition' : exports.ManagementEventRuleCondition,
  'RuleAction' : exports.RuleAction,
  'RuleAction.Microsoft.Azure.Management.Insights.Models.RuleEmailAction' : exports.RuleEmailAction,
  'RuleAction.Microsoft.Azure.Management.Insights.Models.RuleWebhookAction' : exports.RuleWebhookAction
};
```

**Comment by lmazuel** [comment]
@amarzavery yes? I agree this is our situation! Congrats to the NodeJS generator to inherit automatically the fix :)

**Pull request merged**